File: client/src/main/java/com/msc/serverbrowser/data/entites/SampServer.java
package com.msc.serverbrowser.data.entites;

import java.util.Objects;

import com.msc.serverbrowser.util.fx.OneLineStringProperty;

import javafx.beans.property.BooleanProperty;
import javafx.beans.property.IntegerProperty;
import javafx.beans.property.SimpleBooleanProperty;
import javafx.beans.property.SimpleIntegerProperty;
import javafx.beans.property.StringProperty;

@SuppressWarnings("javadoc")
public class SampServer {
	private final BooleanProperty passwordedProperty = new SimpleBooleanProperty();

	private final StringProperty	hostnameProperty		= new OneLineStringProperty();
	private final StringProperty	addressProperty			= new OneLineStringProperty();
	private final StringProperty	actualPlayersProperty	= new OneLineStringProperty();
	private final StringProperty	modeProperty			= new OneLineStringProperty();
	private final StringProperty	languageProperty		= new OneLineStringProperty();
	private final StringProperty	lagcompProperty			= new OneLineStringProperty();
	private final StringProperty	websiteProperty			= new OneLineStringProperty();
	private final StringProperty	versionProperty			= new OneLineStringProperty();
	private final StringProperty	mapProperty				= new OneLineStringProperty();

	private final IntegerProperty	portProperty		= new SimpleIntegerProperty();
	private final IntegerProperty	playersProperty		= new SimpleIntegerProperty();
	private final IntegerProperty	maxPlayersProperty	= new SimpleIntegerProperty();

	public SampServer(final String address, final Integer port) {
		setAddress(address);
		setPort(port);
	}

	public BooleanProperty passwordedProperty() {
		return passwordedProperty;
	}

	public StringProperty hostnameProperty() {
		return hostnameProperty;
	}

	public IntegerProperty portProperty() {
		return portProperty;
	}

	public StringProperty addressProperty() {
		return addressProperty;
	}

	public StringProperty lagcompProperty() {
		return lagcompProperty;
	}

	public StringProperty languageProperty() {
		return languageProperty;
	}

	public IntegerProperty maxPlayersProperty() {
		return maxPlayersProperty;
	}

	public StringProperty modeProperty() {
		return modeProperty;
	}

	public StringProperty playersAndMaxPlayersProperty() {
		return actualPlayersProperty;
	}

	public IntegerProperty playersProperty() {
		return playersProperty;
	}

	public StringProperty versionProperty() {
		return versionProperty;
	}

	public StringProperty websiteProperty() {
		return websiteProperty;
	}

	public StringProperty mapProperty() {
		return mapProperty;
	}

	public boolean isPassworded() {
		return passwordedProperty.get();
	}

	public String getHostname() {
		return hostnameProperty.get();
	}

	public String getAddress() {
		return addressProperty.get();
	}

	public String getLagcomp() {
		return lagcompProperty.get();
	}

	public String getLanguage() {
		return languageProperty.get();
	}

	public Integer getMaxPlayers() {
		return maxPlayersProperty.get();
	}

	public String getMode() {
		return modeProperty.get();
	}

	public Integer getPort() {
		return portProperty.get();
	}

	public Integer getPlayers() {
		return playersProperty.get();
	}

	public String getVersion() {
		return versionProperty.get();
	}

	public String getWebsite() {
		return websiteProperty.get();
	}

	public String getMap() {
		return mapProperty.get();
	}

	public void setPassworded(final boolean passworded) {
		passwordedProperty.set(passworded);
	}

	public void setAddress(final String address) {
		addressProperty.set(address);
	}

	public void setHostname(final String hostname) {
		hostnameProperty.set(hostname);
	}

	public void setPort(final Integer port) {
		portProperty.set(port);
	}

	public void setLagcomp(final String lagcomp) {
		lagcompProperty.set(lagcomp);
	}

	public void setLanguage(final String language) {
		languageProperty.set(language);
	}

	public void setMaxPlayers(final Integer maxPlayers) {
		maxPlayersProperty.set(maxPlayers);
		updatePlayersAndMaxPlayers();
	}

	public void setMode(final String mode) {
		modeProperty.set(mode);
	}

	public void setPlayers(final Integer players) {
		playersProperty.set(players);
		updatePlayersAndMaxPlayers();
	}

	public void setVersion(final String version) {
		versionProperty.set(version);
	}

	public void setWebsite(final String website) {
		websiteProperty.set(website);
	}

	public void setMap(final String map) {
		mapProperty.set(map);
	}

	private void updatePlayersAndMaxPlayers() {
		actualPlayersProperty.set(playersProperty.get() + "/" + maxPlayersProperty.get());
	}

	@Override
	public String toString() {
		return getAddress() + ":" + getPort();
	}

	@Override
	public boolean equals(final Object object) {
		if (Objects.isNull(object) || !object.getClass().equals(SampServer.class)) {
			return false;
		}

		final SampServer compare = (SampServer) object;
		return compare == this || getAddress().equals(compare.getAddress()) && getPort().equals(compare.getPort());
	}

	@Override
	public int hashCode() {
		return (getAddress() + getPort()).hashCode();
	}

}

File: src/main/kotlin/com/msc/serverbrowser/data/entites/SampServer.kt
package com.msc.serverbrowser.data.entites

import javafx.beans.property.*
import com.msc.serverbrowser.util.fx.OneLineStringProperty
import java.util.*

class SampServer(address: String, port: Int) {
    private val passwordedProperty = SimpleBooleanProperty()

    private val hostnameProperty = OneLineStringProperty()
    private val addressProperty = OneLineStringProperty()
    private val actualPlayersProperty = OneLineStringProperty()
    private val modeProperty = OneLineStringProperty()
    private val languageProperty = OneLineStringProperty()
    private val lagcompProperty = OneLineStringProperty()
    private val websiteProperty = OneLineStringProperty()
    private val versionProperty = OneLineStringProperty()
    private val mapProperty = OneLineStringProperty()

    private val portProperty = SimpleIntegerProperty()
    private val playersProperty = SimpleIntegerProperty()
    private val maxPlayersProperty = SimpleIntegerProperty()

    private val lastJoinProperty = SimpleLongProperty()

    var isPassworded: Boolean
        get() = passwordedProperty.get()
        set(passworded) = passwordedProperty.set(passworded)

    var hostname: String
        get() = hostnameProperty.get()
        set(hostname) = hostnameProperty.set(hostname)

    var address: String
        get() = addressProperty.get()
        set(address) = addressProperty.set(address)

    var lagcomp: String
        get() = lagcompProperty.get()
        set(lagcomp) = lagcompProperty.set(lagcomp)

    var language: String
        get() = languageProperty.get()
        set(language) = languageProperty.set(language)

    var maxPlayers: Int?
        get() = maxPlayersProperty.get()
        set(maxPlayers) {
            maxPlayersProperty.set(maxPlayers!!)
            updatePlayersAndMaxPlayers()
        }

    var mode: String
        get() = modeProperty.get()
        set(mode) = modeProperty.set(mode)

    var port: Int
        get() = portProperty.get()
        set(port) = portProperty.set(port)

    var players: Int?
        get() = playersProperty.get()
        set(players) {
            playersProperty.set(players!!)
            updatePlayersAndMaxPlayers()
        }

    var version: String
        get() = versionProperty.get()
        set(version) = versionProperty.set(version)

    var website: String?
        get() = websiteProperty.get()
        set(website) = websiteProperty.set(website!!)

    var map: String
        get() = mapProperty.get()
        set(map) = mapProperty.set(map)

    var lastJoin: Long?
        get() = lastJoinProperty.get()
        set(lastJoin) = lastJoinProperty.set(lastJoin!!)

    init {
        addressProperty.set(address)
        portProperty.set(port)
    }

    fun passwordedProperty(): BooleanProperty {
        return passwordedProperty
    }

    fun hostnameProperty(): StringProperty {
        return hostnameProperty
    }

    fun portProperty(): IntegerProperty {
        return portProperty
    }

    fun addressProperty(): StringProperty {
        return addressProperty
    }

    fun lagcompProperty(): StringProperty {
        return lagcompProperty
    }

    fun languageProperty(): StringProperty {
        return languageProperty
    }

    fun maxPlayersProperty(): IntegerProperty {
        return maxPlayersProperty
    }

    fun modeProperty(): StringProperty {
        return modeProperty
    }

    fun playersAndMaxPlayersProperty(): StringProperty {
        return actualPlayersProperty
    }

    fun playersProperty(): IntegerProperty {
        return playersProperty
    }

    fun versionProperty(): StringProperty {
        return versionProperty
    }

    fun websiteProperty(): StringProperty {
        return websiteProperty
    }

    fun mapProperty(): StringProperty {
        return mapProperty
    }

    fun lastJoinProperty(): LongProperty {
        return lastJoinProperty
    }

    private fun updatePlayersAndMaxPlayers() {
        actualPlayersProperty.set(playersProperty.get().toString() + "/" + maxPlayersProperty.get())
    }

    override fun toString(): String {
        return "$address:$port"
    }

    override fun equals(`object`: Any?): Boolean {
        if (Objects.isNull(`object`) || `object`!!.javaClass != SampServer::class.java) {
            return false
        }

        val compare = `object` as SampServer?
        return compare === this || address == compare!!.address && port == compare.port
    }

    override fun hashCode(): Int {
        return (address + port).hashCode()
    }
}
